snuba/cli/subscriptions.py

```diff
@@ -0,0 +1,195 @@
+import signal
+import uuid
+from concurrent.futures import ThreadPoolExecutor
+from contextlib import closing
+from datetime import timedelta
+from typing import Optional, Sequence
+
+import click
+
+from snuba import settings
+from snuba.datasets.factory import DATASET_NAMES, enforce_table_writer, get_dataset
+from snuba.environment import setup_logging, setup_sentry
+from snuba.redis import redis_client
+from snuba.subscriptions.consumer import TickConsumer
+from snuba.subscriptions.data import PartitionId
+from snuba.subscriptions.executor import SubscriptionExecutor
+from snuba.subscriptions.scheduler import SubscriptionScheduler
+from snuba.subscriptions.store import RedisSubscriptionDataStore
+from snuba.subscriptions.worker import (
+    SubscriptionResultCodec,
+    SubscriptionWorker,
+)
+from snuba.util import create_metrics
+from snuba.utils.codecs import PassthroughCodec
+from snuba.utils.streams.batching import BatchingConsumer
+from snuba.utils.streams.kafka import (
+    CommitCodec,
+    KafkaConsumer,
+    KafkaProducer,
+    build_kafka_consumer_configuration,
+)
+from snuba.utils.streams.synchronized import SynchronizedConsumer
+from snuba.utils.streams.types import Topic
+
+
+@click.command()
+@click.option(
+    "--dataset",
+    "dataset_name",
+    default="events",
+    type=click.Choice(DATASET_NAMES),
+    help="The dataset to target",
+)
+@click.option("--topic")
+@click.option("--partitions", type=int)
```

Comment:
Do you need to specify the str type for the `topic` option or does click infer it from the type hints of the function? If you do not need to specify that, why did you need to specify that on --partitions ?